### PR TITLE
Feature/rework death fx

### PR DESCRIPTION
**Pull request by AlbertVVila**
# Feature Death Animation 
- Now there are 2 death animations, one for normal melee hits and another for skills or criticals
- Added a shot FX for the Bandolero

# Test
- Play the Graveyard scene and watch both animations, check that killing an enemy doesn't crash the game.

### CODE REVIEW

File: Scripts/InventoryScript/InventoryScript.cpp [hunk 1]
```diff
@@ -98,7 +98,7 @@ void InventoryScript::Start()
 	}
 
 	itemDesc = App->scene->FindGameObjectByName("ItemStats");
-	assert(itemDesc != nullptr);
+	//assert(itemDesc != nullptr);
```
Comment (Rubetoman) [inline Scripts/InventoryScript/InventoryScript.cpp]: You left this assert commented.

File: Scripts/PlayerMovement/PlayerMovement.cpp [hunk 2]
```diff
@@ -699,7 +699,7 @@ void PlayerMovement::Update()
 	if (App->time->gameTimeScale == 0) return;
 
 	deltatime = App->time->gameDeltaTime;
-
+	LOG("%d", IsExecutingSkill());
```
Comment (Rubetoman) [inline Scripts/PlayerMovement/PlayerMovement.cpp]: You left some debug code.

File: Scripts/RangeEnemyAIScript/EnemyStateAttack.cpp [hunk 3]
```diff
@@ -41,19 +54,16 @@ void EnemyStateAttack::HandleIA()
...
-	if (timer > enemy->projectileDelay1 && !projShot1)
-	{
-		// Reset projectile position
-		enemy->projectile1->transform->SetPosition(enemy->enemyController->GetPosition() + math::float3(0.f,enemy->projectileScript1->offsetHeight,0.f));
-		enemy->projectile1->transform->SetRotation(enemy->enemyController->GetRotation());
-		enemy->projectileScript1->shooted = true;
-		projShot1 = true;
-		enemy->projectile1->SetActive(true);
-	}
+	//if (timer > enemy->projectileDelay1 && !projShot1)
```
Comment (Rubetoman) [inline Scripts/RangeEnemyAIScript/EnemyStateAttack.cpp]: There are comments that can be deleted.

File: Scripts/InventoryScript/InventoryScript.cpp (hunk 1 above)
Comment (AlbertVVila) [reply Scripts/InventoryScript/InventoryScript.cpp]: crashes with that assert, will delete it, some scenes don't have that gameobject

File: Scripts/PlayerMovement/PlayerMovement.cpp (hunk 2 above)
Comment (AlbertVVila) [reply Scripts/PlayerMovement/PlayerMovement.cpp]: Damn, forgot that one, thanks.

File: Scripts/RangeEnemyAIScript/EnemyStateAttack.cpp (hunk 3 above)
Comment (AlbertVVila) [reply Scripts/RangeEnemyAIScript/EnemyStateAttack.cpp]: 👍

**Pull request merged**